plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -613,6 +612,17 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
    void repost_exhausted_transactions(const fc::time_point& deadline);
    bool push_read_only_transaction(transaction_metadata_ptr trx, next_function<transaction_trace_ptr> next);
 
+   void set_produce_block_offset(uint32_t produce_block_offset_ms) {
+      EOS_ASSERT(produce_block_offset_ms < (config::producer_repetitions * config::block_interval_ms), plugin_config_exception,
+                 "produce-block-offset-ms ${p} must be [0 - ${max})", ("p", produce_block_offset_ms)("max", config::producer_repetitions * config::block_interval_ms));
+      _produce_block_cpu_effort_us = fc::milliseconds( config::block_interval_ms - produce_block_offset_ms / config::producer_repetitions );
+   }
+
+   fc::microseconds get_produce_block_offset() const {
+      return fc::milliseconds( (config::block_interval_ms * config::producer_repetitions) -
+                               ((_produce_block_cpu_effort_us.count()/1000) * config::producer_repetitions) );
```

Comment:
Why not? 
```suggestion
      return fc::milliseconds(produce_block_offset_ms);
```

Reply:
Because `_produce_block_cpu_effort_us` is the offset for each individual block.

Reply:
if you look above, `_produce_block_cpu_effort_us = fc::milliseconds( config::block_interval_ms - produce_block_offset_ms / config::producer_repetitions )`. Just replace in `get_produce_block_offset` and I believe what I suggested is correct.

Reply:
In the existing code, you multiply by `config::producer_repetitions`, so it is *not* for each individual block.

Reply:
Sorry, I'm not following. There is no variable `produce_block_offset_ms`. We could save it for this purpose, but it can be easily calculated. The user setting is for offset of the 12 block round. The internal variable is how much window of time to give an individual block.

Reply:
Oh, my bad, I thought it was a member. Of course you recalculate it here. 
I think it may be more straightforward to save it as a member variable instead of recalculating it.

Reply:
If we keep it separate I'm afraid they may drift. Seems better to have one value of truth.

Reply:
It seems odd to store not the one provided by the user, but the one calculated, but I guess the calculated one is the one we use. So it is fine.